test/functional/treesitter/highlight_spec.lua

```diff
@@ -523,6 +523,37 @@ describe('treesitter highlighting (C)', function()
     }
   end)
 
+  it('supports extending queries', function()
+    insert([[
+    int x = 2;
+    int y = 2;
+    return x + y;
+    ]])
+
+    exec_lua [[
+      vim.treesitter.query.set("c", "highlights", "[\";\"] @variable", { extends = true })
+      vim.treesitter.highlighter.new(vim.treesitter.get_parser(0, "c"))
+    ]]
+
+    screen:expect {
+      grid = [[
+      {5:int} {4:x} {2:=} {3:2}{4:;}                                                       |
+      {5:int} {4:y} {2:=} {3:2}{4:;}                                                       |
+      {2:return} {4:x} {2:+} {4:y;}                                                    |
+      ^                                                                 |
+      {1:~                                                                }|*13
+                                                                       |
+      ]],
+      attr_ids = {
+        [1] = { bold = true, foreground = Screen.colors.Blue },
+        [2] = { bold = true, foreground = Screen.colors.Brown },
+        [3] = { foreground = Screen.colors.Magenta },
+        [4] = { foreground = Screen.colors.Cyan4 },
+        [5] = { foreground = Screen.colors.SlateBlue },
+      },
+    }
+  end)
```

Comment:
In addition, the behavior `vim.treesitter.query.get()` after `vim.treesitter.query.set()` can also be tested.